oxide-vpc/src/engine/dhcpv6.rs

```diff
@@ -0,0 +1,80 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2022 Oxide Computer Company
+
+//! Implements DHCPv6 as supported in the Oxide VPC environment.
+
+use crate::api::VpcCfg;
+use core::num::NonZeroU32;
+use opte::api::Direction;
+use opte::api::Ipv6Addr;
+use opte::api::OpteError;
+use opte::engine::dhcpv6::AddressInfo;
+use opte::engine::dhcpv6::Dhcpv6Action;
+use opte::engine::dhcpv6::LeasedAddress;
+use opte::engine::layer::Layer;
+use opte::engine::port::PortBuilder;
+use opte::engine::port::Pos;
+use opte::engine::rule::Action;
+use opte::engine::rule::HairpinAction;
+use opte::engine::rule::Rule;
+
+cfg_if! {
+    if #[cfg(all(not(feature = "std"), not(test)))] {
+        use alloc::sync::Arc;
+    } else {
+        use std::sync::Arc;
+    }
+}
+
+pub fn setup(
+    pb: &mut PortBuilder,
+    cfg: &VpcCfg,
+    ft_limit: NonZeroU32,
+) -> Result<(), OpteError> {
+    // This layer contains no actions if the client has not been configured with
+    // IPv6 support.
+    let ip_cfg = match cfg.ipv6_cfg() {
+        None => return Ok(()),
```

Comment:
I agree with the larger point. I was trying to say that _this layer_, `"dhcpv6"` has nothing to do. I could have made that clearer though. I like the idea of a first "gateway" layer that drops all outbound IPv6 traffic (or whatever it is, based on the configuration). That could do the same thing inbound, or we could have a similar layer that is first on the inbound direction. It does keep coming up though, we should at least write something up.

Reply:
I meant to ask. Would you prefer I add rules here that explicitly drop all DHCPv6 traffic?

Reply:
I'm planning to drop all DHCPv6 traffic, similar to ICMPv6, when there's no IPv6 at all for the client. I still agree, a blanket rule dropping any EtherType of IPv6 is better, but this will be good in the meantime.